File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletGeolocator/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletGeolocatorExtension: RufletExtension {
  public init() {}

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "Geolocator" ? GeolocatorService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_geolocator/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'geolocator.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "Geolocator":
        return GeolocatorService(control: control);
      default:
        return null;
    }
  }
}
